### PR TITLE
Ent to span

### PR DESCRIPTION
**Pull request by vmenger**
* Store matched entities in `doc.spans['ents']` instead of `doc.ents`
* Make sure entities are not overwritten
* Moved `Term` to a separate class (`clinlp.ie.term`)

### CODE REVIEW

**Comment by vmenger** [comment]
Fixes #36 #38 #78

**Comment by vmenger** [comment]
@bramiozo Zou jij hier een reviewende blik op willen werpen?

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+:exclamation: = Breaking change
+
+## (unreleased)
+
+### Changed
+* :exclamation: Renamed the `clinlp_entity_matcher` to `clinlp_rule_based_entity_matcher`
+* :exclamation: `clinlp` now stores entities in `doc.spans['ents']` rather than `doc.ents`, allowing for overlap
```
Comment (mjhdoeleman) [inline CHANGELOG.md:14]: Contrast "new" conflict solution with previous behaviour (i.e. overlap vs. storing only the latest matched entity)

File: scripts/test_data/generate_qualifier_regression_data.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 import spacy
 
 import clinlp  # noqa: F401
+from clinlp.ie import SPANS_KEY
```
Comment (mjhdoeleman) [inline scripts/test_data/generate_qualifier_regression_data.py:14]: "ents" as constant because it may change in the future?

**Review by mjhdoeleman — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (vmenger) [reply CHANGELOG.md:14]: Thanks, added some clarification here

File: scripts/test_data/generate_qualifier_regression_data.py (hunk 2 above)
Comment (vmenger) [reply scripts/test_data/generate_qualifier_regression_data.py:14]: Yes, if we ever want to change it it's nice to have it in one place only

**Pull request merged**